src/screens/nano/NanoContractsList.js

```diff
@@ -0,0 +1,325 @@
+          }
+          return;
+        }
+      } catch (e) {
+        // Error in request
+        setError(true);
+      } finally {
+        setLoading(false);
+      }
+    },
+    [pagination]
+  );
+
+  /**
+   * Handle load history depending on the query params in the URL
+   * useCallback is needed here because this method is used as a dependency in the useEffect
+   */
+  const handleLoadData = useCallback(async () => {
+    const queryParams = pagination.obtainQueryParams();
+    let after = null;
+    let before = null;
+    // We need to set data here because the user might change the URL directly
+    if (queryParams.sort) {
+      setSort(queryParams.sort);
+    }
+
+    if (queryParams.id) {
+      if (queryParams.page === 'previous') {
+        before = queryParams.id;
+      } else if (queryParams.page === 'next') {
+        after = queryParams.id;
+      } else {
+        // Params are wrong
+        pagination.clearOptionalQueryParams();
+      }
+    }
+
+    if (queryParams.search) {
+      // We need to set it here because the user might add the search param directly in the URL
+      searchRef.current.value = queryParams.search;
+      setIsSearching(true);
+    }
+
+    loadData(after, before, queryParams.search, queryParams.sort);
+  }, [loadData, pagination]);
+
+  useEffect(() => {
+    // Get data when the URL changes
+    handleLoadData();
+  }, [location, loadData, pagination, handleLoadData]);
+
+  /**
+   * Track search field change so we can remove the search key from the URL when the user clears the search field
+   * XXX Unfortunately there's no event for the clear icon click
+   */
+  const handleSearchChange = () => {
+    if (searchRef.current.value === '' && isSearching) {
+      setIsSearching(false);
+      const newURL = pagination.setURLParameters({}, ['search']);
+      navigate(newURL);
+      handleLoadData();
+    }
+  };
+
+  /**
+   * Handle any key up event to execute the search when the user clicks "Enter" key
+   *
+   * @param {Object} e Key up event object.
+   */
+  const handleKeyUp = e => {
+    if (e.key === 'Enter') {
+      setIsSearching(true);
+      // If we are starting a new search, we must ignore the pagination parameters and go to the first page
+      const paramsToDelete = ['id', 'page'];
+      const newURL = pagination.setURLParameters(
+        { search: searchRef.current.value },
+        paramsToDelete
+      );
+      navigate(newURL);
+      handleLoadData();
+    }
+  };
+
+  /**
+   * When the user clicks in the list row, we redirect to the nano contract detail screen
+   *
+   * @param {string} id Blueprint id
+   */
+  const handleClickRow = id => {
+    navigate(`/nano_contract/detail/${id}`);
+  };
+
+  /**
+   * Method to handle click to sort the list.
+   * We clear the pagination parameters but keep the search.
+   */
+  const onSortClicked = () => {
+    let newSort;
+    if (sort && sort === 'asc') {
```

Comment:
suggestion(non-blocking): Using only this strict comparison, `sort` values of `undefined`, `null` or `''` would continue working as intended here.
```suggestion
    if (sort === 'asc') {
```

Reply:
Done https://github.com/HathorNetwork/hathor-explorer/pull/387/commits/af59e9557b32b270b7779620b714746d8d183af9